Editor/EditorCore/DrawShapeTool.cs

```diff
@@ -406,13 +415,29 @@ void OnOverlayGUI(UObject overlayTarget, SceneView view)
             }
         }
 
+        void ResetPrefs()
+        {
+            var type = EditorShapeUtility.availableShapeTypes[s_ActiveShapeIndex];
+            if(currentShapeInOverlay == m_LastShapeCreated)
+                m_LastShapeCreated = null;
+
+            UndoUtility.RegisterCompleteObjectUndo(currentShapeInOverlay, "Change Shape");
+            currentShapeInOverlay.SetShape(EditorShapeUtility.CreateShape(type), currentShapeInOverlay.pivotLocation);
+            SetBounds(currentShapeInOverlay.size);
+
+            ProBuilderEditor.Refresh();
+        }
+
         void DrawShapeGUI()
         {
             if(m_BoldCenteredStyle == null)
                 m_BoldCenteredStyle = new GUIStyle("BoldLabel") { alignment = TextAnchor.MiddleCenter };
 
             EditorGUILayout.LabelField(EditorShapeUtility.shapeTypes[s_ActiveShapeIndex.value], m_BoldCenteredStyle, GUILayout.ExpandWidth(true));
 
+            if(EditorShapeUtility.s_ResetUserPrefs.value)
+                ResetPrefs();
```

Comment:
This looks pretty fragile. If there isn't already a callback when user settings are reset, we should add one and react to events rather than polling.

Reply:
Yeah I agree on this. The other way to do this would be to modify the data structure used to store the current shape prefs.
This way seems less intrusive to me and doesn't need any modification from trunk as well.

Reply:
Ok, let's not block release for this but I think we should refactor this asap